src/object-store/src/manager.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use snafu::ensure;
+
+use crate::error::Result;
+use crate::{error, ObjectStore};
+
+/// Manages multiple object stores so that users can configure a storage for each table.
+/// This struct certainly have one default object store, and can have zero or more custom object stores.
+pub struct ObjectStoreManager {
+    stores: HashMap<String, ObjectStore>,
+    default_object_store: ObjectStore,
+}
+
+impl ObjectStoreManager {
+    pub fn try_new(
+        stores: HashMap<String, ObjectStore>,
+        default_object_store: &str,
+    ) -> Result<Self> {
+        ensure!(
+            stores.keys().any(|key| key == default_object_store),
+            error::DefaultStorageNotFoundSnafu {
+                default_object_store: default_object_store.to_string()
+            }
+        );
+
+        // Safety: We've already checked in the above place whether this exists.
+        let default_object_store = stores.get(default_object_store).unwrap().clone();
+        Ok(ObjectStoreManager {
+            stores,
+            default_object_store,
+        })
+    }
+
+    pub fn find(&self, name: &str) -> Option<ObjectStore> {
+        self.stores.get(name).cloned()
+    }
+
+    pub fn default_object_store(&self) -> ObjectStore {
+        self.default_object_store.clone()
+    }
```

Comment:
nitpick: Sometimes we don't need to clone the object store. What about returning `&ObjectStore`?
```suggestion
    pub fn default_object_store(&self) -> &ObjectStore {
        &self.default_object_store
    }
```

Reply:
But `find` already returns a copy, maybe we should keep `default_object_store()` returning a copy too.

Reply:
I addressed your review: 09fc1d7